neat_ml/utils/figure_utils.py

```diff
@@ -0,0 +1,376 @@
+    std_labels, label_map = _standardise_labels(raw_labels, x_comp)
+    wrapper = GMMWrapper(gmm, x_comp, x_features)
+    x_start, x_stop = map(float, xrange)
+    y_start, y_stop = map(float, yrange)
+    xs = np.linspace(x_start, x_stop, resolution)
+    ys = np.linspace(y_start, y_stop, resolution)
+    xx, yy = np.meshgrid(xs, ys)
+    z_raw = wrapper.predict(np.c_[xx.ravel(), yy.ravel()])
+    z = np.vectorize(label_map.get)(z_raw).reshape(xx.shape)
+
+    if region_colors is None:
+        region_colors = ["aquamarine", "lightsteelblue"]
+
+    if plot_regions:
+        ax.contourf(
+            xx,
+            yy,
+            z,
+            levels=[-0.5, 0.5, 1.5],
+            colors=region_colors,
+            alpha=decision_alpha,
+        )
+
+    ax.contour(
+        xx,
+        yy,
+        z,
+        levels=[0.5],
+        colors=boundary_color,
+        linewidths=3,
+    )
+    proxy_artist = Line2D(
+        [0], 
+        [0], 
+        label="Decision Boundary", 
+        color=boundary_color, 
+        linewidth=3
+    )
+    ax.legend(handles=[proxy_artist])
+
+    boundary_points = extract_boundary_from_contour(z, xs, ys, level=0.5)
+    return gmm, std_labels, boundary_points
+
+def plot_gmm_composition_phase(
+    df: pd.DataFrame,
+    x_col: str,
+    y_col: str,
+    phase_col: str,
+    ax,
+    point_cmap: Optional[Sequence[str]] = None,
+) -> None:
+    """
+    Trains a GMM and creates a scatter trace for data points.
+
+    Optionally creates region traces as well, mimicking the 
+    original script's logic.
+
+    Parameters:
+    -----------
+        df (pd.DataFrame): DataFrame with composition and 
+                           phase data.
+        x_col (str): Name of the column for the x-axis 
+                     component.
+        y_col (str): Name of the column for the y-axis 
+                     component.
+        phase_col (str): Name of the column containing 
+                         phase information.
+        xrange (list[int]): X-axis composition range.
+        yrange (list[int]): Y-axis composition range. 
+        n_components (int): Number of GMM components.
+        random_state (int): Seed for reproducibility.
+        region_colors (Optional[list[str]]): Colors for the 
+                                             phase regions.
+        resolution (int): Grid resolution for any contour plots.
+        plot_regions (bool): Whether to plot the filled regions.
+        point_cmap (Optional[list[str]]): Colormap for the
+                                          scatter points.
+
+    Returns:
+    -------
+        None
+    """
+    x_features = df[phase_col].to_numpy().reshape(-1, 1)
+    gmm = GaussianMixture(n_components=2, random_state=42)
+    raw_labels = gmm.fit_predict(x_features)
+
+    x_comp = df[[x_col, y_col]].to_numpy()
+    std_labels, _ = _standardise_labels(raw_labels, x_comp)
+    if point_cmap is None:
+        point_cmap = ("aquamarine", "lightsteelblue")
```

Comment:
not hit by regression tests:

<img width="485" height="56" alt="Image" src="https://github.com/user-attachments/assets/702cee0f-17e5-44a1-a0f6-6501d2245a2f" />